Automattic/jetpack
# Nav Unification: Support absolute URLs in upsell nudges

Helps fixing https://github.com/Automattic/wp-calypso/issues/57498.

#### Changes proposed in this Pull Request:
The Nav Unification logic was previously assuming that all the upsell nudges links were pointing to Calypso and that the `wordpress.com` domain was missing.

However, there are some upsell nudges that point to WP Admin where the appropriate domain is already part of the URL.

On this PR, we check that the upsell nudge link doesn't include a domain (and thus should point to Calypso) before completing it with `wordpress.com`.

#### Jetpack product discussion
N/A.

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
- Install Jetpack Beta on an WoA dev site and switch to the branch of this PR.
- In your WP.com sandbox, re-enable the Crowdsignal JITM:
  - Revert D69407-code.
  - Adjust the dates set with `within_date_range` to include the current date.
  - If the plan is not newer than 30 days, comment out the `with_plan_newer_than` line.
- Allow incoming HTTP connections in your sandbox: `sudo global-http enable`.
- Connect your WoA dev site to your WP.com sandbox by adding this code to the `wp-config.php` file: `define( 'JETPACK__SANDBOX_DOMAIN', '<YOUR_SANDBOX_DOMAIN>.wordpress.com' )`.
- Activate the Crowdsignal plugin on your WoA Atomic site.
- Go to a WP Admin page.
- The Crowdsignal JITM should show up in the sidebar.
- Make sure the CTA button points to the correct WP Admin page.
- Make sure that other JITMs still point to Calypso correctly.
